### ISSUE TITLE
[BUG] (short description)

### ISSUE THREAD

**Issue by m0rtiZz** [bigtreetech/BIGTREETECH-SKR-mini-E3#173]
On firmware for skr mini e3 1.2 "firmware-bltouch-for-z-homing.bin" i'm inserting xyz offsets with m851 command, saving with m500 comand, and after rebooting the printer offsets are set to default, not these what i entered.

### Steps to reproduce

1. M851 X-44 Y-16 Z-2.8
2. M500
3. Reboot printer

**Expected behavior**
Set my offsets and have them saved

**Actual behavior**
Offsets reseting to default

### Additional Information
using configuration.h provided with marlin downloaded directly from marlin site

**Comment by USAFRichards** [comment]
I have the same problem. Thank you.

**Comment by Thinkersbluff** [comment]
Please confirm whether you are using Marlin 2.0.1.  There is a BUG in 2.0 that corrupts EEPROM saved to flash memory.

**Comment by m0rtiZz** [comment]
> Please confirm whether you are using Marlin 2.0.1. There is a BUG in 2.0 that corrupts EEPROM saved to flash memory.

I'm using firmware created by btt, "firmware-bltouch-for-z-homing.bin", it's marlin 2.0.x

**Comment by Thinkersbluff** [comment]
Closed bug #16532 or #16118 may help, then. 
If you have print counting on, you could try turning that off.
Forcing Marlin to use the SD card as the emulated EEPROM works.
Upgrading Marlin to 2.0.1 works.

**Comment by USAFRichards** [comment]
How do you force Marlin to use the SD card?  

Also, is there a way to modify the "firmware-bltouch-for-z-homing.bin" to make fixes?

I was able to add emulated EEPROM to use the flash memory via the pins file when compiling, but then my stepper motors were louder than with the "firmware-bltouch-for-z-homing.bin".  If I could just modify the pins file of this firmware I would be able to keep all the other settings.

**Comment by Thinkersbluff** [comment]
> How do you force Marlin to use the SD card?

"When I comment-out the line "#define FLASH_EEPROM_EMULATION" in the file pins_BTT_SKR_MINI_E3.h, then Marlin 2.0.x reverts to using the SD card to store EEPROM.dat. In that configuration, the EEPROM does work as expected."

I realize there is a lot to read in closed bug #16532.  The above info is included there.
NOTE also, if you do decide to upgrade to Marlin 2.0.1, from the bugfix branch, there are other edits and issues documented in #16532.

**Comment by USAFRichards** [comment]
I read this and I also removed the SD card, deleted the bin file and reset the machine. Seems to hold.

**Comment by Thinkersbluff** [comment]
Cool!  Does that mean this bug can be closed?

**Comment by USAFRichards** [comment]
Now, if I don’t “Save Settings” before shutting down the machine I lose my offsets. If I perform the “Save Settings” before shutting down they are save for the next boot up.

**Comment by brew99** [comment]
That is how it is suppose to work.  You need to "save settings", either to SD or use emulated EEPROM, then they will be recalled on next boot-up

**Comment by krakow2000** [comment]
> Now, if I don’t “Save Settings” before shutting down the machine I lose my offsets. If I perform the “Save Settings” before shutting down they are save for the next boot up.

But what with PID nozzle and bed if you use repetier or other soft to calibrate?

**Comment by brew99** [comment]
@krakow2000 just use M500 command in repetier to save the settings.   M503 to recall them and make sure they were saved

**Comment by huzargarden** [comment]
Can bigtreetech update the file "firmware-bltouch-for-z-homing.bin" to eliminate this bug? I have the same issue and i can't save the setting on the BTT TFT35 screen

**Comment by brew99** [comment]
Of course they can, but when is the real question!!   Best to take some time and learn how to compile it yourself using the Vanilla Marlin not BTT version

**Comment by huzargarden** [comment]
then what is the point to keeping files with errors  if everyone will go to compile its own version? 

not everyone have time and wants to spent time to config and compile marlin if there is frlirmware file availble to upload. Some poeple here are actually using printers to print stuff and not to consistly improve them and print parts for the printer ;)

**Comment by brew99** [comment]
I totally agree it is not very useful to have a firmware that has such a major issue in it.  Support is not a very strong point for BTT, so sometimes you need to take it into your own hands and correct it yourself

**Comment by Xenon-2510** [comment]
I have this issue with the BTT firmware too, I have put the M851 for Z offset in my Slicer script so it sets the Z offset prior to each print. More of a workaround than a fix but for those who don't want to compile from Vanilla Marlin this will work.
